simulation/include/traccc/simulation/measurement_smearer.hpp

```diff
@@ -96,24 +96,26 @@ struct measurement_smearer {
             if constexpr (std::is_same_v<
                               typename mask_t::local_frame_type,
                               detray::line2D<traccc::default_algebra>>) {
-                iomeas.local0 =
+                iomeas.local0 = static_cast<float>(
```

Comment:
Shouldn't we rather `static_cast` to `traccc::scalar` here instead? The EDM does use `traccc::scalar` and not `float` for a reason. I think here we should be just a bit more careful. :thinking:

Reply:
core EDM uses `traccc::scalar` but IO EDM uses float - I don't recall why the IO EDM happen to be float though

Reply:
Right. I also think most of EDM do not need to be flexible in scalar type as they are now in traccc :thinking: I am also thinking of removing scalar template type for some classes in detray